tests/arch/arc/arc_dsp_sharing/src/load_store.c

```diff
@@ -0,0 +1,239 @@
+
+#if defined(CONFIG_ISA_ARCV2)
+#include "dsp_regs_arc.h"
+#endif
+
+#include "dsp_context.h"
+#include "test_common.h"
+
+/* space for dsp register load/store area used by low priority task */
+static struct dsp_register_set dsp_reg_set_load;
+static struct dsp_register_set dsp_reg_set_store;
+
+/* space for dsp register load/store area used by high priority thread */
+static struct dsp_register_set dsp_reg_set;
+
+static volatile unsigned int load_store_low_count;
+static volatile unsigned int load_store_high_count;
+
+/* Indicates that the load/store test exited */
+static volatile bool test_exited;
+
+/* Semaphore for signaling end of test */
+static K_SEM_DEFINE(test_exit_sem, 0, 1);
+
+/**
+ * @brief Low priority DSP load/store thread
+ *
+ * @ingroup kernel_dspsharing_tests
+ *
+ * @see k_sched_time_slice_set(), memset(),
+ * _load_all_dsp_registers(), _store_all_dsp_registers()
+ */
+static void load_store_low(void)
+{
+	volatile unsigned int i;
+	bool error = false;
+	unsigned char init_byte;
+	unsigned char *store_ptr = (unsigned char *)&dsp_reg_set_store;
+	unsigned char *load_ptr = (unsigned char *)&dsp_reg_set_load;
+	/*
+	 * Initialize dsp load buffer to known values;
+	 * these values must be different than the value used in other threads.
+	 */
+	init_byte = MAIN_DSP_REG_CHECK_BYTE;
+	for (i = 0; i < SIZEOF_DSP_REGISTER_SET; i++) {
+		load_ptr[i] = init_byte++;
+	}
+
+	/* Loop until the test finishes, or an error is detected. */
+	for (load_store_low_count = 0; !test_exited; load_store_low_count++) {
+
+		/*
+		 * Clear store buffer to erase all traces of any previous
+		 * dsp values that have been saved.
+		 */
+		(void)memset(&dsp_reg_set_store, 0, SIZEOF_DSP_REGISTER_SET);
+
+		/*
+		 * Utilize an architecture specific function to load all the
+		 * dsp registers with known values.
+		 */
+		_load_all_dsp_registers(&dsp_reg_set_load);
+
+		/*
+		 * Waste some cycles to give the high priority load/store
+		 * thread an opportunity to run when the low priority thread is
+		 * using the dsp registers.
+		 *
+		 * IMPORTANT: This logic requires that sys_clock_tick_get_32() not
+		 * perform any dsp operations!
+		 */
+		k_busy_wait(100);
+
+		/*
+		 * Utilize an architecture specific function to dump the
+		 * contents of all dsp registers to memory.
+		 */
+		_store_all_dsp_registers(&dsp_reg_set_store);
+		/*
+		 * Compare each byte of buffer to ensure the expected value is
+		 * present, indicating that the dsp registers weren't
+		 * impacted by the operation of the high priority thread(s).
+		 *
+		 * Display error message and terminate if discrepancies are
+		 * detected.
+		 */
+		init_byte = MAIN_DSP_REG_CHECK_BYTE;
+
+		for (i = 0; i < SIZEOF_DSP_REGISTER_SET; i++) {
+			if (store_ptr[i] != init_byte) {
+				TC_ERROR("Found 0x%x instead of 0x%x @ "
+					 "offset 0x%x\n",
+					 store_ptr[i],
+					 init_byte, i);
+				TC_ERROR("Discrepancy found during "
+					 "iteration %d\n",
+					 load_store_low_count);
```

Comment:
Zephyr now allows 100 character per line - so we can avoid some some string splitting:
```
				TC_ERROR("Found 0x%x instead of 0x%x @offset 0x%x\n",
					 store_ptr[i], init_byte, i);
				TC_ERROR("Discrepancy found during iteration %d\n",
					 load_store_low_count);
```